### PR TITLE
feat: Add Sample Data

### PR DESCRIPTION
**Pull request by yangchoo**
Adding sample data to enable easier testing/visualization. Add in
infrastructure of a mocked Connection object for returning sample
responses.

### CODE REVIEW

File: test/sampledata/connection_test.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        )
+        assert (
+            mock_connection.projects()
+            == load_sampledata_json("sample-org-projects.json")["projects"]
+        )
+        assert mock_connection.runs(project_id="p123") == load_sampledata_json(
+            "p123-runs.json"
+        )
+
+    def test_jupyter_project(self):
```
Comment (yangchoo) [inline test/sampledata/connection_test.py:41]: example usage: If this looks good, will roll this out for the other objects such as Run, Container, etc.

**Comment by dzleidig** [comment]
Getting that same CI error when trying to run locally.
```
================================================================================================== ERRORS ===================================================================================================
____________________________________________________________________________ ERROR collecting test/sampledata/connection_test.py ____________________________________________________________________________
ImportError while importing test module '/Users/dylanleidig/dev/transcriptic/test/sampledata/connection_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
test/sampledata/connection_test.py:1: in <module>
    import pandas as pd
E   ModuleNotFoundError: No module named 'pandas'
============================================================================================= warnings summary ==============================================================================================
transcriptic/util.py:44
  /Users/dylanleidig/dev/transcriptic/transcriptic/util.py:44: DeprecationWarning: invalid escape sequence \[
    pattern = "\[(.*?)\]"
```

I think this makes sense, but if you can explain the non-test use case a bit more, that would be good. Like, is our intention that users will used the mocked connection? Or the `sample_data`/`sample_project_attr`? I'm not too familiar with the Jupyter notebooks functionality.

**Comment by yangchoo** [comment]
> if you can explain the non-test use case a bit more, that would be good. Like, is our intention that users will used the mocked connection? Or the `sample_data`/`sample_project_attr`?

Sure. Its a WIP right now, but what I'm thinking is in the Jupyter notebook itself, having examples similar to what's written in the test case here.

**Comment by yangchoo** [comment]
@dzleidig fyi I checked in an example notebook in https://github.com/strateos/transcriptic/pull/174 for easier preview of what this would be used for.

See https://github.com/strateos/transcriptic/blob/22afc190245914a32abdef2b20aea188b5994a02/notebooks/objects.ipynb

**Comment by dzleidig** [comment]
Oh very cool. Pretty cool formatting by Github on that too 😎 .

**Comment by ramu-phanimukla** [comment]
> I.e. they'll initialize a mocked connection and use the library/api just as they would with a real connection.

If we have some feature flag, then by default we shall use mocked connection and objects. Users need not explicitly initialize this mocked connection object.

**Comment by yangchoo** [comment]
> If we have some feature flag, then by default we shall use mocked connection and objects. Users need not explicitly initialize this mocked connection object.

I've changed the interface, let me know if this makes sense?
https://github.com/strateos/transcriptic/blob/c4a05dcce309b48208a2aa4914242bc5afc42891/notebooks/objects.ipynb

**Review by dzleidig — APPROVED** [review]
Nice, I like this. Seems like a good feature-flaggish compromise...using the constructor. I'm going to approve.

**Pull request merged**